### PR TITLE
feat: add post-order export & restoring

### PR DESCRIPTION
**Pull request by cool-develope**
ref: #852

### CODE REVIEW

File: tree_test.go [hunk 1]
```diff
@@ -205,13 +204,14 @@ func TestTree_Build_Load(t *testing.T) {
 
 	// export the tree at version 12,000 and import it into a sql db
 	ctx := context.Background()
+	traverseOrder := PreOrder
```
Comment (yihuang) [inline tree_test.go:207]: Can you add test for post-order traversal.

**Review by yihuang — COMMENTED** [review]
Implementation looks good to me, but I think we need to add some tests for it.

File: tree_test.go (hunk 1 above)
Comment (kocubinski) [reply tree_test.go:207]: Yea, let's have this test run for both pre and post order.

File: snapshot.go [hunk 2]
```diff
@@ -592,6 +514,165 @@ func (snap *sqliteSnapshot) prepareWrite() error {
...
+
+		return node, nil
+	}
+
+	node, err := step()
+
+	return node, uniqueVersions, err
+}
+
+func (snap *sqliteSnapshot) writeSnapNode(node *Node, version int64, count int) error {
```
Comment (kocubinski) [inline snapshot.go:644]: I think the write logic may need to change a bit here now, if you pull `v2-alpha6` you'll see what I mean.  I needed to incorporate an additional flag to accommodate multiple write paths in snapshot.

https://github.com/cosmos/iavl/blob/640a21ec34b532d04886b6de2085c809dca6f343/snapshot.go#L30C1-L31

**Pull request merged**